refnx/reduce/platypusnexus.py

```diff
@@ -626,6 +627,118 @@ def _check_sample_environments(self, d, h5d):
         return d
 
 
+class SpinChannel(Enum):
+    """
+    Describes the spin state of a polarised neutron beam.
+    """
+
+    UPUP = (1, 1)
+    UPDOWN = (1, 0)
+    DOWNUP = (0, 1)
+    DOWNDOWN = (0, 0)
+
+
+class SpinSet(object):
+    """
+    Describes a set of spin-channels at a given angle.
+
+    TODO: implement polarisation efficiency correction within this class
+
+    Parameters
+    ----------
+    list of {str, h5data}
+        list of str, or list of h5py file handles pointing to
+        a set of polarised neutron beam files
+
+    data_folder: {str, Path}
+        Path to the data folder containing the data to be reduced.
+    """
+
+    def __init__(self, dataset, data_folder=None):
```

Comment:
Creating a `SpinSet` requires mandatory UPUP and DOWNDOWN. I think it would be cleaner to have the function signature as:

```
# or down_down, up_up, ...
def __init__(self, up_up, down_down, up_down=None, down_up=None, data_folder=None):
```
because then the SpinSet cannot be created without providing at least two of the spin states.